server/api/resolvers/donationFormResolvers.ts

```diff
@@ -0,0 +1,42 @@
+import { DonationForm, DonationFormInterface } from "../../database/models/donationFormModel";
+import { RequestGroup, RequestGroupInterface } from "../../database/models/requestGroupModel";
+
+const donationFormEmbeddingFromDonationForm = (donationForm: DonationFormInterface) => {
+    return {
+        _id: donationForm._id
+    };
+};
+const addDonationFormToRequestGroup = async (donationForm, requestGroupId) => {
+    const newRequestGroup = await RequestGroup.findById(requestGroupId);
+    newRequestGroup.donationForms.push(donationFormEmbeddingFromDonationForm(donationForm));
+    await newRequestGroup.save();
+};
+
+const donationFormQueryResolvers = {
+    donationForm: async (_, { _id }, ___): Promise<DonationFormInterface> => {
+        return DonationForm.findById(_id).exec();
+    },
+    donationForms: async (_, __, ___): Promise<Array<DonationFormInterface>> => {
+        return DonationForm.find().exec();
+    }
+};
+
+const donationFormMutationResolvers = {
+    createDonationForm: async (_, { donationForm }, ___): Promise<DonationFormInterface> => {
+        const newDonationForm = await new DonationForm({ ...donationForm }).save();
+
+        // link donation group to request group
+        if (newDonationForm.requestGroup) {
+            await addDonationFormToRequestGroup(donationForm, newDonationForm.requestGroup);
+        }
+        return newDonationForm;
+    }
```

Comment:
You'll need an `updateDonationForm` endpoint for the matching page
You can add it in this PR or a future PR